### PR TITLE
Update dependencies and docs

### LINKED ISSUE #24: Compile warning on 0.2.3 on Elixir 1.8.0-otp-21.2.2
Latest released version is 0.2.3 still.

When depender project is compiling with warnings as errors then it fails to build:
```elixir
warning: variable "package" does not exist and is being expanded to "package()", please use parentheses to remove the ambiguity or change the variable name
  /home/<user>/<project>/deps/ueberauth_identity/mix.exs:11

warning: variable "description" does not exist and is being expanded to "description()", please use parentheses to remove the ambiguity or change the variable name
  /home/<user>/<project>/deps/ueberauth_identity/mix.exs:17

warning: variable "deps" does not exist and is being expanded to "deps()", please use parentheses to remove the ambiguity or change the variable name
  /home/<user>/<project>/deps/ueberauth_identity/mix.exs:18

warning: variable "docs" does not exist and is being expanded to "docs()", please use parentheses to remove the ambiguity or change the variable name
  /home/<user>/<project>/deps/ueberauth_identity/mix.exs:19

warning: variable "docs_extras" does not exist and is being expanded to "docs_extras()", please use parentheses to remove the ambiguity or change the variable name
  /home/<user>/<project>/deps/ueberauth_identity/mix.exs:39

==> ueberauth_identity
Compiling 2 files (.ex)
warning: variable "default_options" does not exist and is being expanded to "default_options()", please use parentheses to remove the ambiguity or change the variable name
  lib/ueberauth/strategy/identity.ex:57

warning: Dict.get/2 is deprecated. Use the Map module for working with maps or the Keyword module for working with keyword lists
  lib/ueberauth/strategy/identity.ex:57

warning: Dict.get/3 is deprecated. Use the Map module for working with maps or the Keyword module for working with keyword lists
  lib/ueberauth/strategy/identity.ex:57

Generated ueberauth_identity app
```

### PR DESCRIPTION
**Pull request by lessless**
Should close #24  and #23

### CODE REVIEW

**Comment by lessless** [comment]
@doomspork can you please take a look at this?

File: README.md [hunk 1]
```diff
@@ -120,7 +120,7 @@ If you want to disable that behaviour set the following option in your config:
 
 Depending on the configured url you can initial the request through:
 
-    /auth/identity
+    /auth/identity/callback
```
Comment (yordis) [inline README.md:123]: Why did you change this? The initial request is initialized using `/auth/:identity`

Comment (lessless) [reply README.md:123]: @yordis  as per https://github.com/ueberauth/ueberauth_identity/issues/23#issuecomment-400173333 discussion.
Did I misunderstood it?

Comment (yordis) [reply README.md:123]: and I keep forgetting that it is the `identity` strategy 🤦‍♂️

**Review by yordis — APPROVED** [review]

**Comment by yordis** [comment]
🚀
